src/test/java/io/camunda/testing/assertions/JobAssertTest.java

```diff
@@ -106,6 +108,65 @@ void testHasRetries() throws InterruptedException {
       assertThat(actual).hasRetries(1);
     }
 
+    @Test
+    void testHasAnyIncidents() throws InterruptedException {
+      // given
+      Utilities.deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+      Utilities.startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      client
+          .newThrowErrorCommand(actual.getKey())
+          .errorCode("error")
+          .errorMessage("error occurred")
+          .send()
+          .join();
+
+      // then
+
+      assertThat(actual).hasAnyIncidents();
+    }
+
+    @Test
+    void testHasNoIncidents() throws InterruptedException {
+      // given
+      Utilities.deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+      Utilities.startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+
+      // then
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      assertThat(actual).hasNoIncidents();
+    }
+
+    @Test
+    void testExtractLatestIncident() throws InterruptedException {
+      // given
+      Utilities.deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+      Utilities.startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      client
+          .newThrowErrorCommand(actual.getKey())
+          .errorCode("error")
```

Comment:
🔧 We could also use constants here the same way as in `IncidentAssertTest`. Idem in the other testcases.